vllm/model_executor/layers/sampler.py

```diff
@@ -109,13 +140,19 @@ def forward(
             on_device_tensors = None
 
         # Get the logprobs query results.
-        prompt_logprobs, sample_logprobs = _get_logprobs(
-            logprobs, sampling_metadata, sample_results)
-        return _build_sampler_output(sample_results,
-                                     sampling_metadata,
-                                     prompt_logprobs,
-                                     sample_logprobs,
-                                     on_device_tensors=on_device_tensors)
+        prompt_logprobs = None
+        sample_logprobs = None
+        if not sampling_metadata.skip_sampler_cpu_output:
+            prompt_logprobs, sample_logprobs = _get_logprobs(
+                logprobs, sampling_metadata, sample_results)
```

Comment:
FYI @sroy745 this should solve part of the logprob issue. We can go a step further and allow them to be disabled entirely for speedup. Let's chat offline.

Reply:
Ack. I have a draft PR to disable the generation completely. I will ping you offline with the PR.